### DISCUSSION TITLE
Can't get responses from the GATT Server Example

### DISCUSSION THREAD

**Discussion by dev-sda1** [bluez/bluer#121] [Q&A] [answered]
Hi,

I'm running the GATT Server Example on a Raspberry Pi, and interfacing with it through a BLE App on my phone. The app will pick it up and connect, and I'm able to see the service and send data, however the server doesn't seem to acknowledge that any sort of connection was made, nor output data sent to it.  It also seems to like to frequently get stuck on a pairing loop with my phone, constantly prompting "raspberrypi would like to pair with your iPhone" 

Any ideas? This is the latest main branch with BlueZ 5.50

**Comment by surban** [comment, marked as answer]
This is most likely a problem with the Bluetooth daemon. Could you try with a more recent BlueZ version?

**Reply by dev-sda1** [reply]
Updated to BlueZ 5.60 and my phone now prompts a pin that doesn't show up on the Pi's end, but pressing pair however seems to make it work for a good minute or so before it drops out?

Update: The issue where it was behind with output was just me adding a ``println!()`` in the loop to check what was happening, turns out it messed with the sync.

![Screenshot_20240205_140630](https://github.com/bluez/bluer/assets/43112896/4b908f62-3e83-43f2-b1a9-14cc60e2112a)

**Comment by surban** [comment]
You can register an [agent](https://docs.rs/bluer/latest/bluer/agent/struct.Agent.html) if you need to do pairing. Another option is to change the security of your GATT properties, so that they do not require a secure connection.
